### PR TITLE
Sk/generated html

### PR DESCRIPTION
**Pull request by shaekuronen**
Not ready to be merged yet

### CODE REVIEW

File: src/_scripts/pages/_style-guide.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// this generates html for the style guide / pattern library elements, groups, components, modules
+// an element, group, component, or module needs to have a top-level html element, because $.html() selects the first element, so if there are siblings, will only select the first
+
+import {prefix} from '../utilities/_helpers';
+
+let $styleGuideItems = $('[data-style-guide-item]');
```
Comment (ArrayKnight) [inline src/_scripts/pages/_style-guide.js]: Is there a reason we need both the data attribute and the class?

File: src/_scripts/pages/_style-guide.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+if ($styleGuideItems.length > 0) {
+
+    require('prismjs');
```
Comment (ArrayKnight) [inline src/_scripts/pages/_style-guide.js]: We should move requires to the top and use import instead

File: src/_scripts/pages/_style-guide.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        return s.replace(/&/g, '&amp;')
+            .replace(/"/g, '&quot;')
+            .replace(/</g, '&lt;')
+            .replace(/>/g, '&gt;');
+    };
+
+    $styleGuideItems.each(function(index) {
+        let $item = $(this);
+        let $itemClone = $item.clone();
+        $itemClone.children('label').remove();
```
Comment (ArrayKnight) [inline src/_scripts/pages/_style-guide.js]: This should be more specific, use the style-guide-label class

File: src/_scripts/pages/_style-guide.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        let itemHtml = $itemClone.html();
+        let renderText = escapeHTML(beautifyHTML(itemHtml));
+        let $styleGuideItem = $(`
```
Comment (ArrayKnight) [inline src/_scripts/pages/_style-guide.js]: This should be turned into a private mixin for the style guide page

File: src/_styles/main.vendor.scss [hunk 5]
```diff
@@ -2,6 +2,8 @@
 
 @import './vars';
 
-@import 'node_modules/bootstrap/scss/bootstrap';
-@import 'node_modules/slick-carousel/slick/slick.scss';
-@import 'node_modules/slick-carousel/slick/slick-theme.scss';
\ No newline at end of file
+@import "node_modules/bootstrap/scss/bootstrap";
```
Comment (ArrayKnight) [inline src/_styles/main.vendor.scss]: Would you mind terribly if these were single quotes?

File: src/style-guide/index.pug [hunk 6]
```diff
@@ -23,47 +23,48 @@ block content
...
                         +p-label('Typography')
-                        h1.display-1 Display Headline 1
+                        div(data-style-guide-item, class= prefix('style-guide-item'))
+                            h1.display-1 Display Headline 1
                         h1.display-2 Display Headline 2
                         h1.display-3 Display Headline 3
                         h1.display-4 Display Headline 4
                         h1 Headline 1
                         h2
-                            | Headline 2
```
Comment (ArrayKnight) [inline src/style-guide/index.pug]: Why are trailing spaces being removed?

File: src/_scripts/pages/_style-guide.js (hunk 1 above)
Comment (shaekuronen) [inline src/_scripts/pages/_style-guide.js]: Data attributes for js selection and classes for css. Separation of concerns. We've discussed this before and I'm willing to come up w/ a standardized solution, but using css classes for js selection is not, in my mind, a great solution for long-term maintainability.

File: src/_scripts/pages/_style-guide.js (hunk 2 above)
Comment (shaekuronen) [inline src/_scripts/pages/_style-guide.js]: Yeah, good point

File: src/_scripts/pages/_style-guide.js (hunk 3 above)
Comment (shaekuronen) [inline src/_scripts/pages/_style-guide.js]: Yep, will do

File: src/_scripts/pages/_style-guide.js (hunk 4 above)
Comment (shaekuronen) [inline src/_scripts/pages/_style-guide.js]: Yep, good point

File: src/_styles/main.vendor.scss (hunk 5 above)
Comment (shaekuronen) [inline src/_styles/main.vendor.scss]: Ha, shizzle, yeah that's my preference too

File: src/style-guide/index.pug (hunk 6 above)
Comment (shaekuronen) [inline src/style-guide/index.pug]: Hmm not sure, maybe editor config

image

Do you have trailing spaces?

Comment (shaekuronen) [inline src/style-guide/index.pug]: Hmm, actually, my question is... is there an advantage to trailing spaces?

**Pull request closed without merging**